discojs/discojs-node/src/data/text_loader.ts

```diff
@@ -1,14 +1,15 @@
-import fs from 'node:fs/promises'
 import { data as tfData } from '@tensorflow/tfjs-node'
-
+import fs from 'node:fs'
 import { data } from '@epfml/discojs-core'
 
 export class TextLoader extends data.TextLoader<string> {
-  async loadDatasetFrom (source: string): Promise<data.Dataset> {
-    // TODO sure, good idea to load the whole dataset in memory #irony
-    const content = await fs.readFile(source)
-    const file = new tfData.FileDataSource(content)
-
-    return new tfData.TextLineDataset(file)
+  loadDatasetFrom (source: string): Promise<data.Dataset> {
+    // TODO: reads all the file at once, 
+    // inputting the file path to FileDataSource isn't supported anymore
+    const inputFile = fs.readFileSync(source)
```

Comment:
better to use async `readFile` from 'node:fs/promises', as it allow for the other async methods to run while the file is loading.

```suggestion
    const inputFile = await fs.readFile(source)
```